### ISSUE TITLE
#import <Flutter/Flutter.h> --> ERROR

### ISSUE THREAD

**Issue by oguzhanozdemir** [flutter/flutter#89347]
I updated my flutter version and add some new features in app. But it's not building. I didn't catch it. How can I solve this?

```
`Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running pod install... 1.308ms
Running Xcode build...
Xcode build done. 2,7s
Failed to build iOS app
Error output from Xcode build:
↳
objc[83639]: Class AMSupportURLConnectionDelegate is implemented in both ?? (0x1f061f8f0) and ?? (0x117e542b8). One of the two will be used. Which one is undefined.
objc[83639]: Class AMSupportURLSession is implemented in both ?? (0x1f061f940) and ?? (0x117e54308). One of the two will be used. Which one is undefined.
** BUILD FAILED **

Xcode's output:
↳
In file included from /Users/....[flutterpath]...../flutter233/.pub-cache/hosted/pub.dartlang.org/path_provider-1.6.27/ios/Classes/FLTPathProviderPlugin.m:5:
/Users/....[flutterpath]...../flutter233/.pub-cache/hosted/pub.dartlang.org/path_provider-1.6.27/ios/Classes/FLTPathProviderPlugin.h:5:9: fatal error: 'Flutter/Flutter.h' file
not found
#import <Flutter/Flutter.h>
^~~~~~~~~~~~~~~~~~~
1 error generated.
In file included from /Users/....[flutterpath]...../flutter233/.pub-cache/hosted/pub.dartlang.org/package_info-0.4.3+2/ios/Classes/FLTPackageInfoPlugin.m:5:
/Users/....[flutterpath]...../flutter233/.pub-cache/hosted/pub.dartlang.org/package_info-0.4.3+2/ios/Classes/FLTPackageInfoPlugin.h:5:9: fatal error: 'Flutter/Flutter.h' file
not found
#import <Flutter/Flutter.h>
^~~~~~~~~~~~~~~~~~~
1 error generated.
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.`
```

**Comment by oguzhanozdemir** [comment]
Yes,

I added.
-------------
flutter doctor -v Result
-------------
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.1 20C69 darwin-arm, locale tr-TR)
    • Flutter version 2.2.3 at /Users/....../Development/flutter233
    • Framework revision f4abaa0735 (10 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/....../Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • CBB90793-8027-40A7-9DEC-......... • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
-------------
IOS Podfile


# Uncomment this line to define a global platform for your project
platform :ios, '9.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings.delete 'IPHONEOS_DEPLOYMENT_TARGET'
      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
        '$(inherited)',
        ## dart: PermissionGroup.calendar
        'PERMISSION_EVENTS=0',

        ## dart: PermissionGroup.reminders
        'PERMISSION_REMINDERS=0',

        ## dart: PermissionGroup.contacts
        'PERMISSION_CONTACTS=0',

        ## dart: PermissionGroup.camera
        'PERMISSION_CAMERA=0',

        ## dart: PermissionGroup.microphone
        'PERMISSION_MICROPHONE=0',

        ## dart: PermissionGroup.speech
        'PERMISSION_SPEECH_RECOGNIZER=0',

        ## dart: PermissionGroup.photos
        'PERMISSION_PHOTOS=0',

        ## dart: [PermissionGroup.location, PermissionGroup.locationAlways, PermissionGroup.locationWhenInUse]
        'PERMISSION_LOCATION=0',

        ## dart: PermissionGroup.notification
        # 'PERMISSION_NOTIFICATIONS=0',

        ## dart: PermissionGroup.mediaLibrary
        'PERMISSION_MEDIA_LIBRARY=0',

        ## dart: PermissionGroup.sensors
        'PERMISSION_SENSORS=0'
      ]
    end
  end
end


----------

run result
           .........
[ +136 ms] Could not build the application for the simulator.
[   +3 ms] Error launching application on iPhone 12 Pro Max.
[   +5 ms] ../../Development/flutter233/.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/allocation.dart:47:33: Error: Expected type 'T' to be a valid and
instantiated subtype of 'NativeType'.
[        ]   final int totalSize = count * sizeOf<T>();
[        ]                                 ^
[ +732 ms] "flutter run" took 14.976ms.
[   +6 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:663:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1043:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:196:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:284:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:232:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
           
           
[  +54 ms] ensureAnalyticsSent: 52ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1

**Comment by darshankawar** [comment]
@oguzhanozdemir 
From the log, it's failing due to below error:

```
[ +5 ms] ../../Development/flutter233/.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/allocation.dart:47:33: Error: Expected type 'T' to be a valid and
instantiated subtype of 'NativeType'.
[ ] final int totalSize = count * sizeOf();
[ ] ^
[ +732 ms] "flutter run" took 14.976ms.
[ +6 ms]
```

which is pointing to `ff-0.1.3`. Try to upgrade it to [latest](https://pub.dev/packages/ffi/install) and see if it works.
Check [this](https://github.com/flutter/flutter/issues/76705#issuecomment-785973810) issue comment for reference.

**Comment by oguzhanozdemir** [comment]
No, it's not working. It gives same errors.

**Comment by darshankawar** [comment]
@oguzhanozdemir 
Your `ios/podfile` need to have the helper function `flutter_additional_ios_build_settings(target)` that Flutter uses. If you remove it, then you'll get the error you are seeing.
The podfile should look like this: https://github.com/flutter/flutter/blob/e605b7c2250e803f46017b1d544fc8c297007546/packages/flutter_tools/templates/cocoapods/Podfile-ios-swift

**Comment by oguzhanozdemir** [comment]
@darshankawar 
I deleted ios folder, create it again and I didn't add permissions rows, (move to info.plist), now it is working. But every update project I have this error. I will try your solution now.

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
